### PR TITLE
Fix/toggles and settings refactoring

### PR DESCRIPTION
**Pull request by zhelezkov**
Basically it contains fixes and refactoring which led to this:
- hook `useAddresses` split to two different hooks, `useNotificationChannel` and `useNotificationChannelDappSubscription`. First one allows to control notification channel(or address) globally. Second one is about to control dapp subscription itself. All new hooks are bounded to one specific `addressType`, e.g. wallet or telegram or email, etc..
- fixed toggler -> made them controllable and stateless
- notification mode route pick and web3 logic simplification
- removed old settings
- type fixes

### CODE REVIEW

File: packages/dialect-react-sdk/hooks/useNotificationChannel.ts [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+          (await mutateAddresses())?.find((it) => it.type === addressType) ||
+          null
+        );
+      } finally {
+        setUpdatingAddress(false);
+      }
+    },
+    [
+      address,
+      addressType,
```
Comment (AlekseiPetrovDev) [inline packages/dialect-react-sdk/hooks/useNotificationChannel.ts:123]: I think addressType not needed here as dependencies

File: packages/dialect-react-sdk/hooks/useNotificationChannel.ts [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+      isUpdatingAddress,
+      mutateAddresses,
+      walletsApi.addresses,
```
Comment (AlekseiPetrovDev) [inline packages/dialect-react-sdk/hooks/useNotificationChannel.ts:126]: also create/delete using walletApi object. There we get specific addresses. What's the reason of this?

**Comment by AlekseiPetrovDev** [comment]
looks good to me

File: packages/dialect-react-sdk/hooks/useNotificationChannel.ts (hunk 1 above)
Comment (zhelezkov) [reply packages/dialect-react-sdk/hooks/useNotificationChannel.ts:123]: why, updateAddress function uses it :)

**Review by tsmbl — APPROVED** [review]
lgtm

File: packages/dialect-react-sdk/hooks/useNotificationChannel.ts (hunk 2 above)
Comment (zhelezkov) [reply packages/dialect-react-sdk/hooks/useNotificationChannel.ts:126]: eslint puts deps like this. I prefer to have whole walletApi, replaced!

**Pull request merged**